nasl/capture_packet.c

```diff
@@ -95,6 +95,65 @@ init_capture_device (struct in_addr src, struct in_addr dest, char *filter)
   return ret;
 }
 
+/**
+ * @brief Capture a link layer frame.
+ *
+ * @param[in] bpf bpf handler
+ * @param[in] timeout the timeout
+ * @param[out] sz size of the frame
+ *
+ * @return the link layer frame.
+ */
+struct ether_header *
+capture_next_frame (int bpf, int timeout, int *sz)
+{
+  int len;
+  int dl_len;
+  char *frame = NULL;
+  char *ret = NULL;
+  struct timeval past, now, then;
+  struct timezone tz;
+
+  if (bpf < 0)
+    return NULL;
+
+  dl_len = get_datalink_size (bpf_datalink (bpf));
+  bzero (&past, sizeof (past));
```

Comment:
`bzero` and `bcopy` are deprecated. Use `memcpy` or `memmove` instead of `bcopy` and `memset` instead of `bzero`.